src/ui/components/shared/LoadingScreen.tsx

```diff
@@ -1,86 +1,65 @@
+  const [colorIndex, setColorIndex] = useState(0);
+  const color = colorOptions[colorIndex % colorOptions.length];
 
   useEffect(() => {
-    const timeoutId = setInterval(changeHoverboardColor, 5000);
-    return () => clearInterval(timeoutId);
-  }, [changeHoverboardColor]);
+    const timeoutId = setInterval(() => {
+      setColorIndex(prevIndex => prevIndex + 1);
+    }, 5_000);
 
-  return (
-    <div className={styles.loadingScreenWrapper}>
-      <DefaultViewportWrapper>
-        <div className={styles.loadingScreenWrapper}>
-          <div className="flex flex-col items-center space-y-2">
-            <div className={styles.hoverboardWrapper} onClick={changeHoverboardColor}>
-              <Hoverboard color={hoverboardColor} />
-            </div>
-            {children}
-          </div>
-        </div>
-      </DefaultViewportWrapper>
-    </div>
-  );
-}
+    return () => clearInterval(timeoutId);
+  }, []);
 
-function LoadingScreen({
-  uploading,
-  awaitingSourcemaps,
-  message,
-  secondaryMessage,
-}: PropsFromRedux & { message: string; secondaryMessage?: string }) {
-  const waitingForMessage =
-    awaitingSourcemaps || uploading ? (
-      <span>Uploading {Math.round(uploading?.amount ? Number(uploading.amount) : 0)}Mb</span>
-    ) : (
-      <>
-        <div className={styles.message} dangerouslySetInnerHTML={{ __html: message }} />
-        {secondaryMessage && <div className={styles.secondaryMessage}>{secondaryMessage}</div>}
-      </>
+  let content: ReactNode;
+  if (isAwaitingSourceMaps || uploadingInfo) {
+    content = (
+      <span>
+        Uploading {Math.round(uploadingInfo?.amount ? Number(uploadingInfo.amount) : 0)}Mb
+      </span>
     );
-
-  const showHighRiskWarning = useHighRiskSettingCount() > 0;
+  } else {
+    content = message;
+  }
 
   return (
-    <LoadingScreenTemplate>
+    <>
       <RecordingDocumentTitle />
-      <div className={styles.messageWrapper}>{waitingForMessage}</div>
-      {showHighRiskWarning && (
-        <div className={styles.HighRiskWarning}>
-          You have advanced settings enabled that may negatively affect performance
+      <DefaultViewportWrapper className={styles.LoadingScreen}>
+        <div className={styles.Spacer}></div>
+        <div className={styles.Hoverboard}>
+          <Hoverboard color={color} />
         </div>
-      )}
-    </LoadingScreenTemplate>
+        {content}
+        <div className={styles.Spacer}></div>
+        {showHighRiskWarning && (
+          <div className={styles.HighRiskWarning}>
+            You have advanced settings enabled that may negatively affect performance
+          </div>
+        )}
+      </DefaultViewportWrapper>
+    </>
   );
 }
 
-const connector = connect((state: UIState) => ({
-  uploading: getUploading(state),
-  awaitingSourcemaps: getAwaitingSourcemaps(state),
-}));
-type PropsFromRedux = ConnectedProps<typeof connector>;
+const colorOptions: Array<"blue" | "green" | "red"> = ["blue", "green", "red"];
```

Comment:
`as const` would help you to keep those literal strings without duplicating them in the type:
```suggestion
const colorOptions = ["blue", "green", "red"] as const;
```

It changes the type to be a readonly tuple though so sometimes u might actually not want to use this.

Reply:
Oh nice. The changes in this file were mostly just cleanup so I dig this follow up.